nhais-adaptor/adaptor/outgoing/birth/message_birth_adaptor.py

```diff
@@ -1,52 +1,67 @@
-from fhirclient.models.operationdefinition import OperationDefinition
-
 import adaptor.fhir_helpers.fhir_finders as finders
-from adaptor.fhir_helpers.fhir_creators import ResourceType, ParameterName, OperationName
+from adaptor.fhir_helpers.fhir_creators import ResourceType, ParameterName
 from adaptor.outgoing.common.common_adaptor import create_patient_name, create_patient_address
+from adaptor.outgoing.message_adaptor import MessageAdaptor
 from edifact.outgoing.models.birth.message_birth import MessageSegmentBirthPatientDetails, \
     MessageSegmentBirthRegistrationDetails
+from edifact.outgoing.models.message import MessageBeginning
+
+
+class MessageBirthAdaptor(MessageAdaptor):
+
+    def create_message_beginning(self) -> MessageBeginning:

```

Comment:
The two implementations of `create_message_beginning` only differ by the reference number. Perhaps worth having the implementation in the base class and having it call an abstract get_reference_number method or something?

Reply:
Nice!!! I like that 😄